### PR TITLE
feat(user-auth): Session Implementation

### PR DESCRIPTION
**Pull request by d-e-v-esh**
## #59

This PR adds the functionality of session management on the backend.

### Changes:
 - Because of session implementation, whenever a user registers an account or logs in to an account, a cookie will be sent to them and saved in their browser. Because of that cookie, the user will stay logged in to our website.
 - A new `me` query is added. It will be used by the front-end to fetch the details of the user that is currently logged in.
 - All the key values are replaced by environment variables.
 - Implementation of CORS with dynamic origin.

### Node Packages Added
 - [ioredis](https://www.npmjs.com/package/ioredis)
 - [cors](https://www.npmjs.com/package/cors)
 - [dotenv](https://www.npmjs.com/package/dotenv)
 - [express-session](https://www.npmjs.com/package/express-session)
 - [connect-redis](https://www.npmjs.com/package/connect-redis)

### CODE REVIEW

File: src/resolvers/userResolver.js [hunk 1]
```diff
@@ -83,6 +92,7 @@ const userResolver = {
         };
       }
 
+      req.session.userId = user.id;
```
Comment (CIPHERTron) [inline src/resolvers/userResolver.js:95]: Was this meant to be an if condition?

File: src/types/userType.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ const userType = gql`
   }
 
   type Query {
+    me: User
```
Comment (CIPHERTron) [inline src/types/userType.js]: Just for the sake of best practices and conventional naming methods, can we change the `me` field to something like `currentUser` or something?

File: src/index.js [hunk 3]
```diff
@@ -1,25 +1,73 @@
...
+
+  app.use(
+    session({
+      name: "obfc",
+      store: new RedisStore({
+        client: redis,
+        disableTouch: true,
+      }),
+      cookie: {
+        maxAge: 1000 * 60 * 60 * 24 * 365 * 10, // 10 years
```
Comment (CIPHERTron) [inline src/index.js:44]: What’s the use of maxAge in cookie?

**Review by CIPHERTron — COMMENTED** [review]
Overall, the PR looks good. Just a fee nits that I have addressed below:

File: src/index.js (hunk 3 above)
Comment (d-e-v-esh) [reply src/index.js:44]: It defines how long the cookie will stay active.
It won't be 10 years in production. I just set it for 10 years so it is easy to point out and I can be sure it is working properly. We'll change it before putting it up in `prod`

image

File: src/types/userType.js (hunk 2 above)
Comment (d-e-v-esh) [reply src/types/userType.js]: Ok, I'll update this.

File: src/resolvers/userResolver.js (hunk 1 above)
Comment (d-e-v-esh) [reply src/resolvers/userResolver.js:95]: Here, the `userId` of the user that just logged in is being saved to our session. The same is being done in the `register` mutation.

**Review by CIPHERTron — APPROVED** [review]
LGTM 🎉

**Pull request merged**